### PR TITLE
Set up the application required files

### PR DESCRIPTION
**Pull request by dennisja**
created run.py for running the application
created app/models.py to hold the application models
created app/__init__.py to initialize the application
created app/views.py to hold the applicatioon views
created app/decorators.py to hold the required decorators
created config.py to keep track of the application configurations
created db_create.py to enable database creation
created db_upgrade.py to handle database upgrading
created db_migrate.py to haandle database migrations
created db_downgrade.py to handle database downgrades
created tests.py to handle application unit tests
created yummy.log for loggind some application specific information

### CODE REVIEW

**Comment by dennisja** [comment]
@patrickcmd, @matthewacha my pull request

**Comment by PatrickCmd** [comment]
I see your trying to implement databases. Please take note on Challenge 2 NO.16 and the self-assessment guideline of this challenges. For now data structures(lists, dictionaries etc) and OOP concepts are needed to implement this challenge.
Please take note.
The UI design is great

File: app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+#initialize app
+app = Flask(__name__)
+app.config.from_object("config")
+db = SQLAlchemy(app)
+
+from app import views, models
```
Comment (daisyndungu) [inline app/__init__.py]: Add a new line at the end of this file

File: config.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import os 
+
+basedir = os.path.abspath(os.path.dirname(__file__))
+# path to the database
+SQLALCHEMY_DATABASE_URI = 'sqlite:///'+os.path.join(basedir,"app.db")
```
Comment (matthewacha) [inline config.py:5]: use "app_db"or "app" instead of  "app.db" , it automatically creates the database and adds .db

File: run.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from app import app
+
+if __name__ == "__main__":
```
Comment (matthewacha) [inline run.py:3]: remove line 3, its redundant

**Comment by matthewacha** [comment]
check the files you created, i left comments on specific lines

File: config.py (hunk 2 above)
Comment (dennisja) [reply config.py:5]: Thanks @matthewacha

File: app/__init__.py (hunk 1 above)
Comment (dennisja) [reply app/__init__.py]: Ok, i will @daisyndungu

**Pull request merged**